apps/mull-ui/src/app/pages/home/discover/discover-page.tsx

```diff
@@ -1,34 +1,20 @@
-import { gql, useQuery } from '@apollo/client';
 import { ISerializedEvent } from '@mull/types';
 import React from 'react';
+import { useDiscoverEventsQuery } from '../../../../generated/graphql';
 import { EventCard } from '../../../components';
 import '../home-discover.scss';
 
-interface DiscoverData {
-  discoverEvents: Partial<ISerializedEvent>[];
-}
-
-export const GET_DISCOVER_EVENTS = gql`
-  query DiscoverEvents($discoverEventsUserId: Int!) {
-    discoverEvents(userId: $discoverEventsUserId) {
-      id
-      title
-      description
-      startDate
-      endDate
-    }
-  }
-`;
-
 export const DiscoverPage = ({ history }) => {
-  const { data } = useQuery<DiscoverData>(GET_DISCOVER_EVENTS, {
+  const { data } = useDiscoverEventsQuery({
     fetchPolicy: 'network-only',
     // TODO: dynamically pass current UserId
     variables: { discoverEventsUserId: 1 },
   });
 
   if (data) {
-    const events: Partial<ISerializedEvent>[] = data.discoverEvents;
+    const events: Partial<ISerializedEvent>[] = (data.discoverEvents as unknown) as Partial<
+      ISerializedEvent
+    >[];
```

Comment:
You've done this a couple of times where you explicitly mention what the type of the variable is. Since we're casting it, it's not necessary.

```suggestion
   const events = (data.discoverEvents as unknown) as Partial<ISerializedEvent>[];
```

As a side question, is there a way to avoid the double casting being done in a few places?